### DISCUSSION TITLE
Longpress Button to activate encoders in loop

### DISCUSSION THREAD

**Discussion by geoiak** [tttapa/Control-Surface#722] [Q&A]
Thank you for the amazing work you have done!
I have 8 encoders with push switches which i have hooked up on my leonardo to use as a midi pedal board. Everything works with your library. However, I want the encoders to work only if a "long press" of switch number 1 is active. Can you please point me to the right direction to use the if function to activate encoders sending signal inside the loop based on a condition?

**Comment by tttapa** [comment]
All MIDI elements have `enable()` and `disable()` functions, e.g. [`CCRotaryEncoder::enable()`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d62/classCCRotaryEncoder.html#a486f22824bd83c5308a0d70ffac6f758).  
You can use the [`MultiPurposeButton`](https://tttapa.github.io/Control-Surface-doc/Doxygen/db/d45/3_8MultiPurposeButton_8ino-example.html) class to detect long presses.

**Comment by geoiak** [comment]
Thank you for the response!

Best regards
Georgios Iakovou

Chemist, MSc
